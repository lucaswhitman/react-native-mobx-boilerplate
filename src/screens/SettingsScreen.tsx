import React, { Component } from 'react';
import {
  Text,
  View
} from 'react-native';
import { translate } from '../i18n';
import styles from '../styles';

export const SETTINGS_SCREEN_VIEW_CONTAINER = 'SettingsScreenViewContainer';
export const SETTINGS_SCREEN_TEXT_1 = 'SettingsScreenText1';

interface ISettingsScreenProps {}

export default class SettingsScreen extends Component<ISettingsScreenProps> {
  public render(): JSX.Element {
    return (
      <View testID={SETTINGS_SCREEN_VIEW_CONTAINER} style={styles.container}>
        <Text testID={SETTINGS_SCREEN_TEXT_1}style={styles.heading}>
          {translate('common.settings')}
        </Text>
      </View>
    );
  }
}
