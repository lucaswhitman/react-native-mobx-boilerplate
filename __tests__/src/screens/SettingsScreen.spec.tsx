import React from 'react';
import { render, shallow } from 'react-native-testing-library';
import SettingsScreen, { SETTINGS_SCREEN_TEXT_1 } from '../../../src/screens/SettingsScreen';

describe('Settings Screen', () => {
  // snapshot
  it('shallow rendering', () => {
    const { output } = shallow(<SettingsScreen />);

    expect(output).toMatchSnapshot();
  });

  // unit
  it('renders correctly', () => {
    // const { getByTestId, queryByTestId } = render(<SettingsScreen />);
    const { getByTestId } = render(<SettingsScreen />);
    const component = getByTestId(SETTINGS_SCREEN_TEXT_1);

    expect(component.props.children).toEqual('Settings');
  });
});

