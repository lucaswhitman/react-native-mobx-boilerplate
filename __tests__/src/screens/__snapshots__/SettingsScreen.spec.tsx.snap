// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Screen shallow rendering 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="SettingsScreenViewContainer"
>
  <Text
    style={
      Object {
        "fontFamily": "Helvetica",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
    testID="SettingsScreenText1"
  >
    Settings
  </Text>
</View>
`;
